### PR TITLE
Modified Dark Matter Jfactor Computation and Dark Matter Tutorial

### PR DESCRIPTION
**Pull request by katrinstreil**
**Description**

PR to change the wrong integration along the line of sight for the Dark Matter J-Factor computation. 
Added Rectangular Region in the J-fact map in the Tutorial to allow the comparison to the cited paper.  
solves: issue [2769](https://github.com/gammapy/gammapy/issues/2769)

### CODE REVIEW

**Comment by registerrier** [comment]
Thanks @katrinstreil .
We have a code style check in CI. We advise to setup pre-commit hooks in order to enforce the PEP8 code style
`pre-commit install`
Then each time you commit your changes, the `flake8` and `black` will be checked and applied.

File: gammapy/astro/darkmatter/profiles.py [hunk 1]
```diff
@@ -34,11 +34,15 @@ def scale_to_local_density(self):
...
         self.parameters["rho_s"].value *= scale
 
-    def _eval_squared(self, radius):
-        """Squared density at given radius."""
-        return self(radius) ** 2
+    def _eval_squared(self, radius, separation):
+        """Squared density at given radius together with the substitution part"""
+        return (
+            self(radius) ** 2
+            * radius
```
Comment (jdjuvsla) [inline gammapy/astro/darkmatter/profiles.py:41]: Is there a factor 2 missing? As in 2* r * rho^2 / denominator

File: gammapy/astro/darkmatter/utils.py [hunk 2]
```diff
@@ -26,19 +27,36 @@ def __init__(self, geom, profile, distance):
         self.profile = profile
         self.distance = distance
 
-    def compute_differential_jfactor(self):
+    def compute_differential_jfactor(self, ndecade=1e4):
         r"""Compute differential J-Factor.
 
         .. math::
             \frac{\mathrm d J}{\mathrm d \Omega} =
             \int_{\mathrm{LoS}} \mathrm d r \rho(r)
```
Comment (jdjuvsla) [inline gammapy/astro/darkmatter/utils.py:35]: This should read \rho(r)**2. 
I know, it's nothing to do with the changes done here, but when we first review it, we can just correct this as well, I think.

**Review by jdjuvsla — CHANGES_REQUESTED** [review]
Great to see these corrections being made to the J-factor computations. I think there is a factor of 2 missing in the computation, though (see my detailed comments).

File: gammapy/astro/darkmatter/profiles.py (hunk 1 above)
Comment (katrinstreil) [reply gammapy/astro/darkmatter/profiles.py:41]: Yes you are right, there is a factor 2 missing, thanks :)

File: gammapy/astro/darkmatter/utils.py (hunk 2 above)
Comment (katrinstreil) [reply gammapy/astro/darkmatter/utils.py:35]: I made the correction

File: gammapy/astro/darkmatter/tests/test_utils.py [hunk 3]
```diff
@@ -33,5 +37,6 @@ def test_dmfluxmap(jfact):
         jfact * diff_flux.integral(energy_min=energy_min, energy_max=energy_max)
     ).to("cm-2 s-1")
     actual = int_flux[5, 5]
-    desired = 1.9483e-12 / u.cm**2 / u.s
+    print(actual)
```
Comment (registerrier) [inline gammapy/astro/darkmatter/tests/test_utils.py:40]: Please remove print statement.

File: gammapy/astro/darkmatter/utils.py [hunk 4]
```diff
@@ -55,17 +55,19 @@ def compute_differential_jfactor(self, ndecade=1e4):
                     ndecade,
                 )
             )
-            for _ in rmin.flatten()
+            for _ in rmin.ravel()
         ]
+        jfact = u.Quantity(val).to("GeV2 cm-5").reshape(rmin.shape)
+        print(type(jfact))
```
Comment (registerrier) [inline gammapy/astro/darkmatter/utils.py]: remove print statement

**Review by registerrier — APPROVED** [review]
Thanks @katrinstreil .

I have left two inline comments (basically remove the print statements)

File: gammapy/astro/darkmatter/tests/test_utils.py (hunk 3 above)
Comment (katrinstreil) [reply gammapy/astro/darkmatter/tests/test_utils.py]: Thanks, @registerrier. I removed the print statements

**Comment by registerrier** [comment]
The CI fail is unrelated. Merging now. Thanks @katrinstreil for the PR and @jdjuvsla for the review.

@katrinstreil please use commit sign-off for future PRs. See: https://docs.gammapy.org/1.0/development/intro.html#acceptation-of-the-developer-certificate-of-origin-dco

**Pull request merged**